internal/services/hdinsight/hdinsight_hadoop_cluster_resource_test.go

```diff
@@ -2020,3 +2042,258 @@ resource "azurerm_hdinsight_hadoop_cluster" "test" {
+    name                       = "AllowSyncWithAzureAD"
+    priority                   = 101
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "443"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+
+  security_rule {
+    name                       = "AllowRD"
+    priority                   = 201
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "3389"
+    source_address_prefix      = "CorpNetSaw"
+    destination_address_prefix = "*"
+  }
+
+  security_rule {
+    name                       = "AllowPSRemoting"
+    priority                   = 301
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "5986"
+    source_address_prefix      = "AzureActiveDirectoryDomainServices"
+    destination_address_prefix = "*"
+  }
+
+  security_rule {
+    name                       = "AllowLDAPS"
+    priority                   = 401
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "636"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+}
+
+resource azurerm_subnet_network_security_group_association "test" {
+  subnet_id                 = azurerm_subnet.test.id
+  network_security_group_id = azurerm_network_security_group.test.id
+}
+
+resource "azuread_group" "test" {
+  display_name = "AAD DC Administrators %s"
+  description  = "Test for delegating group to administer Azure AD Domain Services"
+}
+
+data "azuread_domains" "test" {
+  only_initial = true
+}
+
+resource "azuread_user" "test" {
+  user_principal_name = "acctestAADDSAdminUser-%s@${data.azuread_domains.test.domains.0.domain_name}"
+  display_name        = "acctestAADDSAdminUser-%s"
+  password            = "TerrAform321!"
+}
+
+resource "azuread_group_member" "test" {
+  group_object_id  = azuread_group.test.object_id
+  member_object_id = azuread_user.test.object_id
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "acctestUAI-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+data "azurerm_subscription" "primary" {}
+
+resource "azurerm_role_assignment" "test" {
+  scope                = data.azurerm_subscription.primary.id
+  role_definition_name = "HDInsight Domain Services Contributor"
+  principal_id         = azurerm_user_assigned_identity.test.principal_id
+}
+
+resource "azurerm_active_directory_domain_service" "test" {
```

Comment:
Whilst it should be fine to include an `azurerm_active_directory_domain_service` in this test, we'll probably have to think about scheduling and parallelization between this test and the test in the `domainservices` package - it's only possible to have a single domain service at any one time and they take quite a while to stand up and tear down.

Reply:
To avoid re-work, may i ask is there any suggestion from your side?

Reply:
I've updated code for your concern.

Reply:
Ah i see. Unfortunately I don't think we can consolidate the tests into one package - the tests really need to live with their respective resources.

I'm looking at the logistics of having a long-running domain service to try and compensate for the concurrency issue, but still working out how to make that work nicely with the domainservices tests themselves.

Reply:
OK. I separated them into their respective resources first.